speech_cleanup.qmd

```diff
@@ -0,0 +1,120 @@
+```{python}
+import pandas_gbq
+import os
+import pandas as pd
+import seaborn as sns
+import numpy as np
+import re
+```
+
+```{python}
+
+os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = "token/gcp_token.json"
+
+df = pandas_gbq.read_gbq("""
+                    SELECT  *
+                    FROM `singapore-parliament-speeches.prod_mart.mart_speeches`
+                         WHERE member_name != ''
+                         AND member_name != 'Speaker'
+                         AND is_primary_question = false
+                    """)
+```
+
+```{python}
+[i for i in df.query('count_speeches_words<50').speech_text]
+
+```
+
+Some visuals, most speeches are within 2000 words with serious outliers
+
+```{python}
+sns.histplot(x = 'count_speeches_words', data = df)
+```
+
+What types of speeches?
+
+```{python}
+
+sns.histplot(y = 'topic_type_name', data = df)
+
+```
+
+What do short speeches look like?
+
+```{python}
+perc_10 = np.percentile(df.count_speeches_words, 10)
+[i for i in df.query(f'count_speeches_words<{perc_10}').speech_text]
+
+```
+
+What do different types of speeches look like?
+
+```{python}
+
+[i for i in df.query('topic_type_name.str.contains("Bill Introduced")').speech_text]
+
+[i for i in df.query('topic_type_name.str.contains("Correction by Written Statements")').speech_text]
+
+```
+
+Some patterns revealed
+1. Many procedural points - 'Yes please', 'Aye', 'last qns', etc. These are all very short and by speaker
+2. empty open brackets
+3. responses to speaker - 'thank you', etc.
+4. there are questions, end with ? and not classified as qns
+5. there are a number of vernacular speeches 
+6. We can omit bill introductions - these are procedural and serves merely to introduce the speech
+7. We can also omit written corrections. These are corrections circulated to correct factual errors in previous speeches.
+8. Both of these constitute only a very small proportion of total speeches
+
+Procedural speeches don't have a particular logic to them and it is tedious to catch them all. Maybe a better way would be to remove all speeches below a certain length.
+Say we assume that a speech is 140 words per minute, we exclude all speeches that are less than half a minute in length, so 70.
+
+In addition we only include speeches within the 95th percentile to remove large outliers
```

Comment:
i've created a flag that identifies short speeches (less than 10th percentile number of words). i suggest we can filter our speeches which belong to this.

however, i disagree that overly long speeches should be excluded from the count. i've created a flag for long speeches (more than 90th percentile number of words). this can be used for analysis, but i will not propose filtering these out.

Reply:
No need to filter out, flag is fine. Either this or the operation is performed in python when I pull from gbq. However I think we can go higher than 4 words. 5 word speeches are equally uninformstive

Reply:
Maybe 0.7 feels like we are cutting out much, but speeches are arbitrarily defined to begin with

Reply:
okay ☺️ i can add more flags. for now, i've done 10th and 90th percentile of the count of number of words.